src/net/socket.rs

```diff
@@ -142,7 +142,10 @@ impl Socket {
             match self.recv_from(time) {
                 Ok(UdpSocketState::MaybeMore) => continue,
                 Ok(UdpSocketState::MaybeEmpty) => break,
-                Err(e) => error!("Encountered an error receiving data: {:?}", e),
+                Err(e) => error!(
+                    self.config.logger,
```

Comment:
Having to specify this each time is a bit annoying. Is there anything clever we could do about that?

Reply:
It'd be possible to make a trait that `Self` implements that returns a logger, but that's about it.